lib/configuration.js

```
@@ -15,150 +15,430 @@
- * @property {Boolean} reportUncaughtExceptions=true - the flag denoting
- *  whether or not to catch and report uncaught errors to the API
- * @property {Object} serviceContext - the hosting applications service
- *  information
- * @property {String} serviceContext.service - the hosting applications service
- *  that it is running atop of
- * @property {String} serviceContext.version - the hosting applications version
- * @property {String} key - an API key used to communicate with the service
- * @property {String} version - the version of the module
- * @property {Boolean} shouldReportErrorsToAPI - a flag indicating whether or
- *  not the Middleware should send errors to the Stackdriver Error Reporting
- *  API. This flag is derived from the `NODE_ENV` environmental varaible being
- *  set to production: if the flag is set to 'production' then the
- *  `shouldReportErrorsToAPI` property will be set to true and the API client
- *  will attempt to send errors to the Stackdriver Error Reporting API.
+ * The _gatherLocalConfiguration function is responsible for determining
+ * directly determing whether the properties `reportUncaughtExceptions` and
+ * `key`, which can be optionally supplied in the runtime configuration, should
+ * be merged into the instance. This function also calls several specialized
+ * environmental variable checkers which not only check for the optional runtime
+ * configuration supplied values but also the processes environmental values.
+ * @memberof Configuration
+ * @private
+ * @function _gatherLocalConfiguration
+ * @returns {Undefined} - does not return anything
  */
-
+Configuration.prototype._gatherLocalConfiguration = function() {
+  if (isPlainObject(this._givenConfiguration)) {
+    if (isBoolean(this._givenConfiguration.reportUncaughtExceptions)) {
+      this._reportUncaughtExceptions = this._givenConfiguration
+        .reportUncaughtExceptions;
+    }
+    if (isString(this._givenConfiguration.key)) {
```

Throw instead of ignoring when the key isn't a string.